### PR TITLE
[NETBEANS-138] Removed InjectedTasksSupportTest test.

### PR DESCRIPTION
**Pull request by blackleg**
Removed InjectedTasksSupportTest test in parsing.indexing module.

The order that the indexers are run is different that the test expect.

It closes this issue [NETBEANS-138](https://issues.apache.org/jira/browse/NETBEANS-138) and old bug [Bugzilla: 237764](https://netbeans.org/bugzilla/show_bug.cgi?id=237764).

### CODE REVIEW

File: parsing.indexing/test/unit/src/org/netbeans/modules/parsing/impl/indexing/InjectedTasksSupportTest.java [hunk 1]
```diff
@@ -359,21 +379,26 @@ public int getIndexVersion() {
 
     private static final class CIHandler extends Handler {
 
+        private static final Logger LOG = Logger.getLogger(CIHandler.class.getName());
 
         enum Event {
             ENQUEUE,
             EXEXUTE
```
Comment (matthiasblaesing) [inline parsing.indexing/test/unit/src/org/netbeans/modules/parsing/impl/indexing/InjectedTasksSupportTest.java]: Please correct the spelling here - I assume "EXECUTE" is meant here.

File: parsing.indexing/test/unit/src/org/netbeans/modules/parsing/impl/indexing/InjectedTasksSupportTest.java [hunk 2]
```diff
@@ -359,21 +379,26 @@ public int getIndexVersion() {
...
+            LOG.info(String.format("Current expected events: %s", this.expectedEvents));
+            LOG.info(String.format("New events: %s", Arrays.toString(events)));
             expectedEvents.addAll(Arrays.asList(events));
+            LOG.info(String.format("Updated expected events: %s", this.expectedEvents));
         }
 
         @Override
         public synchronized void publish(LogRecord record) {
             final String message = record.getMessage();
+            LOG.info(String.format("Record message: %s", message));
```
Comment (matthiasblaesing) [inline parsing.indexing/test/unit/src/org/netbeans/modules/parsing/impl/indexing/InjectedTasksSupportTest.java]: This should expand the message with the parameters. I needed to look twice to see why I saw {0} in the output.

**Comment by blackleg** [comment]
@matthiasblaesing I'm not sure but I think it tests the possibility to extends a indexer with a callable. 
```java
private static final class CustomIndexerImpl extends CustomIndexer {

        private static final Logger LOG = Logger.getLogger(CustomIndexerImpl.class.getName());

        private final Callable<?> preAction;
        private final Callable<?> postAction;

        public CustomIndexerImpl(
                @NullAllowed final Callable<?> preAction,
                @NullAllowed final Callable<?> postAction) {
            this.preAction = preAction;
            this.postAction = postAction;
            LOG.info(String.format("%s, %s", preAction, postAction));
        }

        @Override
        protected void index(Iterable<? extends Indexable> files, Context context) {
            try {
                if (preAction != null) {
                    preAction.call();
                }
                final IndexingSupport is = IndexingSupport.getInstance(context);
                for (Indexable file : files) {
                    LOG.info(String.format("Indexing file. MimeType: %s", file.getMimeType()));
                    final IndexDocument doc = is.createDocument(file);
                    final FileObject fo = URLMapper.findFileObject(file.getURL());
                    doc.addPair(KEY_NAME, fo.getName(), true, true);
                    doc.addPair(KEY_EXT, fo.getExt(), true, true);
                    is.addDocument(doc);
                }
                if (postAction != null) {
                    postAction.call();
                }
            } catch (Exception ioe) {
                Exceptions.printStackTrace(ioe);
            }
        }

    }
```

**Comment by matthiasblaesing** [comment]
@sdedic could you please have a look at this? According to the history you were the last one touching this code and I still don't see the intention of the unittest.

**Comment by sdedic** [comment]
Sorry, I don't have much clue. My last commit was a simple test infrastructure refactoring, had nothing common with the tests' semantic. @tzezula is IMHO better person to ask.

**Comment by matthiasblaesing** [comment]
@blackleg as noone seems to know what this test should test I suggest to remove it. We gain nothing by doing a fix that can't be verified. In other cases I voted to keep the test, as the test itself was fine, but only failed because files were not donated. In this case the test itself is the problem. Are you willing to do that?

**Comment by blackleg** [comment]
@matthiasblaesing It seems right, I will update pull request to remove the test.

**Comment by blackleg** [comment]
@matthiasblaesing I removed the test.

**Pull request merged**